### DISCUSSION TITLE
Error: myCursor is not defined

### DISCUSSION THREAD

**Discussion by LGmatrix13** [prisma/prisma#6435] [Q&A] [answered]
Hello everyone,

I'm trying to add cursor-based pagination. Here is my code. I took after the example found in the [docs](https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination).

**Error:**
`ReferenceError: myCursor is not defined`
**Code:**

```ts
import prisma from '../../../lib/prisma'

export default async function handle(Id, res) {
  const secondQueryResults = prisma.post.findMany({
    take: Id,
    skip: 1,
    cursor: {
      id: myCursor,
    },
    orderBy: {
      id: 'asc',
    },
  })
  
const lastPostInResults = secondQueryResults[3] // Remember: zero-based index! :)
const myCursor = lastPostInResults.id // Example: 5
  res.send(secondQueryResults)
}
```

I should also mention this is a nextjs API route. Any suggestions? Thank you in advance!!

**Comment by ryands17** [comment, marked as answer]
@LGmatrix13 👋

In this case, the best way to do this is passing the `id` via the request in `handle` and then your query would look like this:

```ts
import prisma from '../../../lib/prisma'

export default async function handle(req, res) {
  const secondQueryResults = prisma.post.findMany({
    take: req.Id,
    skip: req.cursor ? 1 : undefined,
    cursor: req.cursor ? { id: myCursor } : undefined,
    orderBy: {
      id: 'asc',
    },
  })

res.send(secondQueryResults)
}
```

The `id` of the last item in the response will always be sent by the frontend and with that you can perform cursor based pagination in the following manner.

**Reply by LGmatrix13** [reply]
Okay. I am trying to do this as a dynamic API route. Is there any way to receive the id from API's URL and use that in `take`, `skip`, and `cursor`? Also, when I make a request with this, it results in an empty object. Is this what is intended?

Thanks for your help!

**Reply by ryands17** [reply]
Yes you can use Next.js [dynamic api routes](https://nextjs.org/docs/api-routes/dynamic-api-routes).

As for getting an empty object, that is only possible if you do not have any data. By default the first `n` records will be specified by what you pass in `take`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
